File: app/src/main/java/com/example/msimple/data/model/ITopicModel.java
package com.example.msimple.data.model;

import com.example.msimple.data.vos.CurrentVO;
import com.example.msimple.data.vos.TopicVO;

import java.util.List;

public interface ITopicModel {

    void getTopic(Response response);

    interface Response{

        void onSuccess(List<TopicVO> topicVOS);
        void onFail(String message);

    }

}

File: app/src/main/java/com/example/msimple/data/model/ITopicModel.kt
package com.example.msimple.data.model

import com.example.msimple.data.vos.CurrentVO
import com.example.msimple.data.vos.TopicVO

interface ITopicModel {

    fun getTopic(response: Response)

    interface Response {

        fun onSuccess(topicVOS: List<TopicVO>)
        fun onFail(message: String)

    }

}
